substrate/primitives/consensus/slots/src/lib.rs

```diff
@@ -91,6 +91,11 @@ impl Slot {
 		Slot(timestamp.as_millis() / slot_duration.as_millis())
 	}
 
+	/// The first timestamp of the slot. Returns None if would overflow for given SlotDuration.
+	pub fn first_timestamp(&self, slot_duration: SlotDuration) -> Option<Timestamp> {
```

Comment:
```suggestion
	/// The timestamp of the slot.
	///
	/// Returns `None` if would overflow for given `SlotDuration`.
	pub fn to_timestamp(self, slot_duration: SlotDuration) -> Option<Timestamp> {
```

I'd rather change the api to this (Slot is Copy)

Reply:
Or maybe just call it `timestamp`. Not really required to have any longer name.

Reply:
I've renamed to `timestamp`, but kept "first" in the documentation, because it returns the first timestamp of the slot